tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/TypeAliasSubstitution.scala

```diff
@@ -133,7 +142,16 @@ object ConstSubstitution {
           if (fun.isDefinedAt(name)) {
             (fun(name), true)
           } else {
-            (tp, false)
+            // The behavior for an OLD_TYPE_ALIAS is to understand it as an uninterpreted type.
+            // Since in the old syntax, we cannot distinguish between uninterpreted types and type aliases,
+            // we cannot do much here.
```

Comment:
I wrote the previous comment, and up until this line it seemed to be the case, from reading the docs, that we support both syntaxes. In particular https://github.com/informalsystems/apalache/blob/ecd5b7188c211b8b7e3f6440f7ecefda4daa420d/tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/TypeAliasSubstitution.scala#L8
mentions `ENTRY` or `$entryInCamlCase`.

But these lines imply that we don't support old aliases anymore. This is a bit confusing and it should be much clearer whether we support old aliases or not (and in what way). Semantically, I don't think treating caps types as uninterpreted means that we support "old aliases" (and the Scala variable names should reflect this).

Reply:
What I meant by "we cannot do much here" is that we cannot detect an error. I think I will fix the comment.